src/utils/walk.ts

```diff
@@ -84,7 +82,20 @@ function processNode(node: JSONSchema4): SchemaNode | void {
     } as IRefNode;
   }
 
-  if (combiner !== undefined) {
+  if (combiner === undefined) {
+    const base: IBaseNode = {
+      id: assignId(node),
+      primaryType: getPrimaryType(node),
+      type: node.type || inferType(node),
```

Comment:
Why have two properties? When will a node ever have a defined type AND a primary type?

Reply:
Any reason we can't just update `inferType` and be done with it?

Reply:
`primaryType` - this props drives the rendering logic
`type` -  displayed in UI

Reply:
Does primary type need to be store on the node if not used in the ui? Can we just call `getPrimaryType` in renderSchema when its needed

Reply:
Fair enough!